domain/src/main/java/org/oppia/domain/analytics/LogReportStorageHelper.kt

```diff
@@ -0,0 +1,82 @@
+package org.oppia.domain.analytics
+
+import androidx.lifecycle.LiveData
+import kotlinx.coroutines.CoroutineDispatcher
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.launch
+import org.oppia.app.model.EventLog
+import org.oppia.app.model.EventLogs
+import org.oppia.data.persistence.PersistentCacheStore
+import org.oppia.util.data.AsyncResult
+import org.oppia.util.data.DataProviders
+import org.oppia.util.logging.Logger
+import org.oppia.util.threading.BackgroundDispatcher
+import javax.inject.Inject
+import javax.inject.Singleton
+
+@Singleton
+class LogReportStorageHelper @Inject constructor(
+  private val cacheStoreFactory: PersistentCacheStore.Factory,
+  private val persistentCacheStore: PersistentCacheStore<EventLogs>,
+  private val dataProviders: DataProviders,
+  private val logger: Logger,
+  @BackgroundDispatcher private val backgroundCoroutineDispatcher: CoroutineDispatcher
+) {
+  private val eventLogStore =
+    cacheStoreFactory.create("event_logs", EventLogs.getDefaultInstance())
+  private val coroutineScope = CoroutineScope(backgroundCoroutineDispatcher)
+  private val storeSizeLimit = 10
+
+  fun addEventLog(eventLog: EventLog) {
+    coroutineScope.launch { checkStoreCacheStatus() }
+    eventLogStore.storeDataAsync(updateInMemoryCache = true) {
+      it.toBuilder().addEventLog(eventLog).build()
+    }.invokeOnCompletion {
+      it?.let {
+        logger.e(
+          "DOMAIN",
+          "Failed to store event log",
+          it
+        )
+      }
+    }
+  }
+
+  private suspend fun checkStoreCacheStatus() {
+    val storeSize = persistentCacheStore.readDataAsync().await().serializedSize.toByte()
+    if (storeSize / 1048576 > storeSizeLimit) {
+      removeEvent(getLeastRecentEvent())
+      checkStoreCacheStatus()
+    }
+  }
+
+  private suspend fun getAllOptionalEvents(): List<EventLog> =
+    eventLogStore.readDataAsync().await().eventLogList
+      .filter { it.priority == EventLog.Priority.OPTIONAL }
+
+  private suspend fun getLeastRecentEvent(): EventLog? =
+    getAllOptionalEvents().minBy { it.timestamp } ?: eventLogStore.readDataAsync()
+      .await().eventLogList.minBy { it.timestamp }
+
+  private suspend fun removeEvent(eventLog: EventLog?) =
+    eventLogStore.readDataAsync().await().eventLogList.remove(eventLog)
+
+  private suspend fun removeEvents(eventLogList: MutableList<EventLog>) =
+    eventLogStore.readDataAsync().await().eventLogList.removeAll(eventLogList)
+
```

Comment:
Please remember that we need to make this work for crash/exception logs too, All of these methods need to support exception logs

Reply:
Yes, modified.